### ISSUE TITLE
Convert nightly build scripts from legacy to dagster

### ISSUE THREAD

**Issue by jdangerx** [catalyst-cooperative/pudl#2296]
The CLI commands which are used in the nightly build scripts have changed a little bit so we need to update the scripts. Changes:

- The clobber arguments no longer exist for all of the commands because dagster just replaces the tables instead of deleting the whole database.
- No more bypass local cache?
- pudl_etl not runs the cems pipeline so no need to run epacems_to_parquet

Scope:
- [ ] local_pudl_etl.sh works in Docker
- [ ] gcp_pudl_etl.sh publishes valid outputs in intake.catalyst-coop

Next steps:
- [x] run local_pudl_etl.sh in a docker instance on a GCE VM

**Comment by jdangerx** [comment]
scope creep alert: while we're in here, it might be worth investigating how we could have the GH action open a PR from `dev` to `main` every time the nightly builds pass. Maybe timebox that to 2 hours.

**Comment by bendnorman** [comment]
I realize this is semi-dependent on #2301 because the nightly builds expects the current `pudl-work` directory structure.

**Comment by jdangerx** [comment]
@bendnorman what would block us from using `"${CONTAINER_PUDL_IN}/data"` as `PUDL_CACHE`  and  `CONTAINER_PUDL_OUT` as `PUDL_OUTPUT`? In my mind that would let the ETL happen but then also have the output/analysis tests run against the Old World?

Separately, why do we `--bypass-local-cache` in GCE? It seems like it might be nice to dump a bunch of stuff into `${PUDL_IN}/data` so that we don't have to download every time, but I'm sure there was some reason to do this.

**Comment by jdangerx** [comment]
My thoughts for getting this working:

* edit `local_pudl_etl.sh`
* edit `docker/.env` to point at etl_fast 🐇 
* build docker image locally with compose
* run something like `docker compose exec pudl-etl conda run ....` where the .... is what's in the build-deploy-pudl.yml, minus the slack/AWS secrets
  * make sure I have the right env to actually auth to GCP
* see if it works, if not, go back to step 1

@bendnorman does that sound right?

**Comment by bendnorman** [comment]
@jdangerx you're right we can just use `"${CONTAINER_PUDL_IN}/data"` as `PUDL_CACHE` and `CONTAINER_PUDL_OUT` as `PUDL_OUTPUT`. We must need to remember to remove the `sqlite` and `parquet` paths from the `copy_outputs_to_intake_bucket()` function.

I think I added `--bypass-local-cache` to make it clear the nightly build pulls from the GCS cache each time. We currently aren't mounting a volume to the container in the VM so we aren't retaining the cache for each run.

You plan sounds good to me!

**Comment by bendnorman** [comment]
Just a heads up, `docker/docker-compose.yml` expects a service account key to use the gcs cache. We could remove the GCS cache flags and authentication function from `local_pudl_etl.sh` so folks don't have to muck with GCP service accounts.

**Comment by bendnorman** [comment]
Also, my janky trick in `gcp_pudl_etl.sh` using `PIPESTATUS` to check if the etl passed or failed probably won't work now that dagster creates multiple processes. @jdangerx do you think it makes sense to convert `gcp_pudl_etl.sh` to a python script that can kick off the dagster jobs and check the result status?

**Comment by jdangerx** [comment]
I think a Python script would be nicer for sure, but in my mind `PIPESTATUS` would fail if our call to `execute_job` fails - and I would *expect* that to fail if any single subprocess fails, but I definitely haven't confirmed that.

**Comment by jdangerx** [comment]
PIPESTATUS seems to be working fine given the various failures we've been getting notified about in slack :)

In terms of "open a PR from `dev` to `main` automatically on success" - I think the easiest thing to do is to just send a link to the ["make new PR" page](https://github.com/catalyst-cooperative/pudl/compare/main...dev) in the success notification if it's a `dev` build.
